File: yh_dual/src/yh_dual_int.cpp
#include "ros/ros.h"
#include "yh_dual/YhDual.h"

void msgCallback(const yh_dual::YhDual::ConstPtr& msg)
{
    ROS_INFO("%d", msg->data);
}

int main(int argc, char** argv)
{
    ros::init(argc, argv, "yh_dual_int");
    ros::NodeHandle nh;

    ros::Subscriber sub = nh.subscribe("yh_dual_topic", 100, msgCallback);

    ros::spin();

    return 0;
}

File: yh_dual/scripts/yh_dual_int.py
#!/usr/bin/python3
# -*- coding: utf-8 -*-
import rospy
from yh_dual.msg import YhDual

def msgCallback(msg):
    rospy.loginfo(msg.data)

def yh_dual_int():
    rospy.init_node("yh_dual_int")
    rospy.Subscriber("yh_dual_topic", YhDual, msgCallback, queue_size=10)
    rospy.spin()

if __name__ == "__main__":
    yh_dual_int()